ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,209 @@
+            )
+            raise
+
+    def query(self, query):
+        """get the prometheus query result
+
+        Args:
+            query(str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        Get alert by alert anem if it's an active alert
+
+        Args:
+             alert(str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict: dictiionary containing alert name as key and alert data as value
+        """
+        response = self.alerts()
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        return alert_list
+
+    def get_firing_alerts(self, alert_name):
+        """
+        get all the firing alerts from list of active alerts
+        """
+        alert_list = self.get_alert(alert_name=alert_name)
+        firing_alerts = []
+        for alert in alert_list:
+            if alert["state"] == "firing":
+                firing_alerts.append(alert)
+
+        return firing_alerts
+
+    def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for an alert and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout(int): wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.get_firing_alerts,
+            alert_name=alert_name,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in firing state.")
+                    for item in sample:
+                        LOGGER.info(f"alert metrics: {item}")
+                    return
+
+        except TimeoutExpiredError:
+            LOGGER.error(f"{alert_name} currently not in firing state")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int: scrape time interval
+        """
+
+        response = self._get_response(query=f"{self.api_v1}/targets")
+        result = response["data"]["activeTargets"]
+        for item in result:
+            if item and item["labels"]["job"] == "prometheus-k8s":
+                scrape_interval = item["scrapeInterval"]
+                scrape_interval = int((re.match(r"\d+", scrape_interval)).group())
+                break
+        return scrape_interval
```

Comment:
you may ended with exception here when `scrape_interval` is not defined.

Reply:
setting scrape_interval to 30 in else condition